engine/src/main/java/io/zeebe/engine/processing/bpmn/task/ReceiveTaskProcessor.java

```diff
@@ -49,45 +47,35 @@ public void onActivating(final ExecutableReceiveTask element, final BpmnElementC
   }
 
   @Override
-  public void onActivated(final ExecutableReceiveTask element, final BpmnElementContext context) {
-    // nothing to do here
-  }
+  public void onComplete(final ExecutableReceiveTask element, final BpmnElementContext context) {
 
-  @Override
-  public void onCompleting(final ExecutableReceiveTask element, final BpmnElementContext context) {
     variableMappingBehavior
         .applyOutputMappings(context, element)
         .ifRightOrLeft(
             ok -> {
               eventSubscriptionBehavior.unsubscribeFromEvents(context);
-              stateTransitionBehavior.transitionToCompleted(context);
+              final var completed = stateTransitionBehavior.transitionToCompleted(context);
+              stateTransitionBehavior.takeOutgoingSequenceFlows(element, completed);
             },
             failure -> incidentBehavior.createIncident(failure, context));
   }
 
   @Override
-  public void onCompleted(final ExecutableReceiveTask element, final BpmnElementContext context) {
-    stateTransitionBehavior.takeOutgoingSequenceFlows(element, context);
-    stateBehavior.removeElementInstance(context);
-  }
+  public void onTerminate(final ExecutableReceiveTask element, final BpmnElementContext context) {
 
-  @Override
-  public void onTerminating(final ExecutableReceiveTask element, final BpmnElementContext context) {
     eventSubscriptionBehavior.unsubscribeFromEvents(context);
-    stateTransitionBehavior.transitionToTerminated(context);
-  }
-
-  @Override
-  public void onTerminated(final ExecutableReceiveTask element, final BpmnElementContext context) {
-    eventSubscriptionBehavior.publishTriggeredBoundaryEvent(context);
     incidentBehavior.resolveIncidents(context);
-    stateTransitionBehavior.onElementTerminated(element, context);
-    stateBehavior.removeElementInstance(context);
-  }
 
-  @Override
-  public void onEventOccurred(
-      final ExecutableReceiveTask element, final BpmnElementContext context) {
-    eventSubscriptionBehavior.triggerBoundaryOrIntermediateEvent(element, context);
+    eventSubscriptionBehavior
+        .findEventTrigger(context)
+        .ifPresentOrElse(
+            eventTrigger -> {
+              final var terminated = stateTransitionBehavior.transitionToTerminated(context);
+              eventSubscriptionBehavior.activateTriggeredEvent(terminated, eventTrigger);
```

Comment:
I am not sure about this comment, but I strongly believe that also in this case we need to call `stateTransitionBehavior.onElementTerminated(element, terminated);`. What do you think?

Reply:
If a boundary event is triggered then we don't need to inform the parent. The parent callback is used to terminate the parent if all child instances are terminated. In this case, the boundary event is activated in the same flow scope. So, the parent can't be terminated. 

Since the termination was caused by the boundary event and not the parent, there is also no reason to inform the parent (at least for now).